mathesar_ui/src/stores/table-data/constraints.ts

```diff
@@ -1,19 +1,28 @@
-import { writable, get as getStoreValue } from 'svelte/store';
-import { getAPI, States } from '@mathesar/utils/api';
+import { writable, get as getStoreValue, derived } from 'svelte/store';
+import {
+  deleteAPI,
+  getAPI,
+  postAPI,
+  States,
+} from '@mathesar/utils/api';
 import type {
   Writable,
   Updater,
   Subscriber,
   Unsubscriber,
+  Readable,
 } from 'svelte/store';
 import type { PaginatedResponse } from '@mathesar/utils/api';
 import type { CancellablePromise } from '@mathesar/components';
 import type { DBObjectEntry } from '@mathesar/App.d';
+import type { Column } from './columns';
+
+export type ConstraintType = 'foreignkey' | 'primary' | 'unique' | 'check' | 'exclude';
 
 export interface Constraint {
   id: number,
   name: string,
-  type: string,
+  type: ConstraintType,
   columns: string[],
```

Comment:
It would be more optimal to make columns a set, we seem to be require finding if a column exists, in several places.

```suggestion
  columns: Set<Column['name']>,
```